lib/paperclip/io_adapters/uri_adapter.rb

```diff
@@ -28,6 +31,13 @@ def cache_current_values
       @size = @content.size
     end
 
+    def extract_attachment_filename
+      if @content.meta.has_key? "content-disposition"
```

Comment:
It should, the adapter uses the open uri module which always contains the meta hash